### PR TITLE
test(page types): Add integration tests for page types

### PR DESCRIPTION
**Pull request by xrutayisire**
## Context

- Completes DT-1375

## The Solution

- Create a first set of integration tests, with each "test()" not relying on implementation details.
- Create helpers to centralize implementation details.
- Fix an error in console when deleting a custom type, ul should not be children of p

## Checklist before requesting a review
- [x] I hereby declare my code ready for review.
- [ ] If it is a critical feature, I have added tests.
- [x] The CI is successful.
- [x] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/test/__testutils__/index.tsx [hunk 1]
```diff
@@ -37,7 +44,10 @@ function render(
       </ThemeUIThemeProvider>
     );
   }
-  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
+  return {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/test/__testutils__/index.tsx:51]: can we add a return type to this function?

File: packages/slice-machine/test/__testutils__/index.tsx [hunk 2]
```diff
@@ -23,6 +24,12 @@ function render(
     ...renderOptions
   }: RenderArgs = {}
 ) {
+  if (!document.getElementById("__next")) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/test/__testutils__/index.tsx:31]: guess that makes sense, as it should only effect some next dependent parts of the code base

File: packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    cwd,
+  });
+
+  await manager.telemetry.initTelemetry({
+    appName: pkg.name,
+    appVersion: pkg.version,
+  });
+  await manager.plugins.initPlugins();
+
+  ctx.msw.use(
```
Comment (MarcMcIntosh) [inline packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx:297]: hmm, trying to figure out if there's a way to spy on these requests,
... maybe on the mock adapter

File: packages/slice-machine/src/features/customTypes/customTypesTable/convertCustomToPageType.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { toast } from "react-toastify";
+
+import { CustomType } from "@prismicio/types-internal/lib/customtypes";
+import { CustomTypeFormat } from "@slicemachine/manager";
+import { managerClient } from "@src/managerClient";
+import { CUSTOM_TYPES_MESSAGES } from "../customTypesMessages";
+
+export async function convertCustomToPageType(
```
Comment (MarcMcIntosh) [inline packages/slice-machine/src/features/customTypes/customTypesTable/convertCustomToPageType.ts]: This kind of looks like the sort of logic that would go into a saga ?
if so, I would recommend using `expectSaga` to test it, that way you can assert side-effects

...it kind of already uses one 🤔

**Review by MarcMcIntosh — COMMENTED** [review]
Good effort,
I have a few questions and observations that I left some comments where appropriate, 
an other one would be that there's few assertions going on, which may or may not be easier to implement

File: packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx (hunk 3 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx:297]: Don't get this comment, here it's kind of a way to spy on them, no? 
We are mocking what's happening normally here

File: packages/slice-machine/src/features/customTypes/customTypesTable/convertCustomToPageType.ts (hunk 4 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/customTypesTable/convertCustomToPageType.ts]: We don't want to directly test this function, that's the goal of the kind of test I'm bringing.
You don't test implementation details. This function is by design an implementation details. If you want to change this function into 2 or 3, users will not see any difference and so your test should not break.

File: packages/slice-machine/test/__testutils__/index.tsx (hunk 1 above)
Comment (xrutayisire) [reply packages/slice-machine/test/__testutils__/index.tsx:51]: Done 👍

File: packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx (hunk 3 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx:297]: msw is network interception, what I'm trying to figure out is what sort of side effects would we be able to assert actually happened. I'll probably need to take some time with code base to ind out what a viable approach would be.

File: packages/slice-machine/src/features/customTypes/customTypesTable/convertCustomToPageType.ts (hunk 4 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/src/features/customTypes/customTypesTable/convertCustomToPageType.ts]: ok

**Review by MarcMcIntosh — APPROVED** [review]

File: packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx (hunk 3 above)
Comment (xrutayisire) [reply packages/slice-machine/src/features/customTypes/__tests__/CustomTypesTablePage.test.tsx:297]: Oh I see, yes In the future we could try to see that but for my use case I reuse Angelo work for that and it's working great :)

**Pull request merged**